internal/commands/setup_test.go

```diff
@@ -68,6 +74,62 @@ func TestSetupNoFail(t *testing.T) {
 	assert.NotPanics(t, func() { Setup(&cobra.Command{}) })
 }
 
+func handler(w http.ResponseWriter, r *http.Request) {
+	_, _ = fmt.Fprintf(w, "Received %s\n", path.Base(r.URL.Path))
+}
+
+type handlerTransport struct{ h http.Handler }
+
+var payload = []byte(`
+apiVersion: qbec.io/v1alpha1
+kind: EnvironmentMap
+spec:
+  environments:
+    prod2:
+      server: https://prod-server
+      includes:
+      - service2
+      properties:
+        envType: prod`)
+
+func (t handlerTransport) RoundTrip(req *http.Request) (*http.Response, error) {
+	resp := &http.Response{
+		Status:     "200 OK",
+		StatusCode: 200,
+		Body:       ioutil.NopCloser(bytes.NewBuffer(payload)),
+		Request:    req,
+	}
+	return resp, nil
+}
+
+func TestListRemoteEnv(t *testing.T) {
+	s := newScaffold(t)
+	defer s.reset()
+	model.HTTPClient().Transport = handlerTransport{http.HandlerFunc(handler)}
```

Comment:
exposing a method called `HTTPClient()` out of the model package just to do this is, er, gross :)

Reply:
just use `httptest` to run a real server in the test and return the response over the network as will be done for real instead of doing this mocking thing.

Reply:
All good,  I'll add httptest and remove the mock.